app/models/manager_refresh/save_collection/helper.rb

```diff
@@ -50,9 +50,29 @@ def create_or_update_inventory!(inventory_collection, record_index, association)
       unique_index_keys = inventory_collection.manager_ref_to_cols
 
       association.find_each do |record|
-        # TODO(lsmola) the old code was able to deal with duplicate records, should we do that? The old data still can
-        # have duplicate methods, so we should clean them up. It will slow up the indexing though.
-        record_index[inventory_collection.object_index_with_keys(unique_index_keys, record)] = record
+        index = inventory_collection.object_index_with_keys(unique_index_keys, record)
+        if record_index[index]
+          # We have a duplicate in the DB, destroy it
+          record_for_destruction = nil
+
+          if record.respond_to?(:service)
```

Comment:
Though similar thing could be achieved by always taking the oldest record? That would work without any specific naming of relations.